azext_iot/central/params.py

```diff
@@ -38,20 +38,10 @@ def load_central_arguments(self, _):
     """
     Load CLI Args for Knack parser
     """
-    with self.argument_context("iot central app") as context:
+    with self.argument_context("iot central") as context:
```

Comment:
Some of these still aren't nested quite right.

I'd ask to organize them by command subgroup rather than specific command, and make sure to collect all of the ones that are required.

```
with self.argument_context("iot central user create") -> with self.argument_context("iot central user"):
    # nest all user related params here

with  self.argument_context("iot central device create") -> with self.argument_context("iot central device"):
    # nest all device params here

etc
```

It might take a bit of trial and error, but this will really help organize the params into logical groups, and reduce risk of param collisions

Reply:
e.g. these params are currently nested to `central`, but they could be more specifically to just `user` and `api-token`

```

        context.argument(
            "assignee",
            options_list=["--user-id", "--assignee"],
            help="ID associated with the user. ",
        )

        context.argument(
            "token_id",
            options_list=["--token-id", "--tkid"],
            help="Unique ID for the API token. ",
        )
```

Reply:
You also put 2 device create's in the same file

```


    with self.argument_context("iot central device create") as context:
        context.argument(
            "device_name",
            options_list=["--device-name"],
            help="Human readable device name. Example: Fridge",
        )


    with self.argument_context("iot central device create") as context:
        context.argument(
            "instance_of",
            options_list=["--instance-of"],
            help="Central template id. Example: urn:ojpkindbz:modelDefinition:iild3tm_uo",
        )
        context.argument(
            "simulated",
            options_list=["--simulated"],
            arg_type=get_three_state_flag(),
            help="Add this flag if you would like IoT Central to set this up as a simulated device. "
            "--instance-of is required if this is true",
        )
```

Reply:
nit - if you want you can put this under a "deprecated" function like in the other files. Up to you


    with self.argument_context("iot central app monitor-events") as context:
        context.argument("timeout", arg_type=event_timeout_type)
        context.argument("properties", arg_type=event_msg_prop_type)
        context.argument(
            "module_id", options_list=["--module-id", "-m"], help="Iot Edge Module ID",
        )

Reply:
can consolidate most of the params for the various diagnostics to a single diagnostics params list

```
with self.argument_context("iot central diagnostics monitor-events") as context:
with self.argument_context("iot central diagnostics validate-messages") as context:
```